### PR TITLE
Equivalence exclusion

### PR DESCRIPTION
**Pull request by Spartakovic**
Add exclusion of content equivalence based on provided uris

### CODE REVIEW

File: src/main/java/org/atlasapi/equiv/generators/EquivalenceGenerators.java [hunk 1]
```diff
@@ -1,30 +1,41 @@
...
+    public EquivalenceGenerators(Iterable<? extends EquivalenceGenerator<T>> generators, Set<String> excludedUris) {
         this.generators = ImmutableList.copyOf(generators);
+        this.excludedUris = excludedUris;
     }
     
     public List<ScoredCandidates<T>> generate(T content, ResultDescription desc) {
         desc.startStage("Generating equivalences");
         Builder<ScoredCandidates<T>> generatedScores = ImmutableList.builder();
-        
+
```
Comment (MartinsIrbe) [inline src/main/java/org/atlasapi/equiv/generators/EquivalenceGenerators.java]: whitespace

File: src/test/java/org/atlasapi/equiv/generators/EquivalenceGeneratorsTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    @Test
+    public void testGeneratorExcludesUriFromTheList() {
+        Item item = new Item();
+        item.setCanonicalUri("excluded");
+        List<ScoredCandidates<Content>> generated = generators.generate(item, resultDescription);
+        assertTrue(generated.isEmpty());
+    }
+
+    @Test
+    public void testGeneratorNotExcludesUriFromTheList() {
```
Comment (Dalamar42) [inline src/test/java/org/atlasapi/equiv/generators/EquivalenceGeneratorsTest.java]: Naming. This tests that the generator does not exclude a uri that is *not* on the exclusion list

File: src/test/java/org/atlasapi/equiv/generators/EquivalenceGeneratorsTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Before
+    public void setUp() {
+        excludedUris = ImmutableSet.of("excluded");
+        generators = EquivalenceGenerators.from(Collections.singleton(generator), excludedUris);
+        when(generator.generate(any(Content.class), any(ResultDescription.class))).thenReturn(candidates);
+    }
+
+    @Test
+    public void testGeneratorExcludesUriFromTheList() {
```
Comment (Dalamar42) [inline src/test/java/org/atlasapi/equiv/generators/EquivalenceGeneratorsTest.java]: Naming: `generatorCreatesNoCandidatesForUriFromExclusionList`

File: src/test/java/org/atlasapi/equiv/generators/EquivalenceGeneratorsTest.java (hunk 2 above)
Comment (Dalamar42) [inline src/test/java/org/atlasapi/equiv/generators/EquivalenceGeneratorsTest.java]: `generatorCreatesCandidatesForNonExcludedUri`

**Pull request merged**